### PR TITLE
ci: add publish workflow and check if package version is updated when needed

### PR DESCRIPTION
**Pull request by froblesmartin**

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+          RELEASE_TAG=$(npm pkg get version | tr -d \")
+          echo "RELEASE_TAG=$RELEASE_TAG" >> $GITHUB_ENV
+
+      - name: Generate GitHub tag and release
+        if: steps.changed-files.outputs.any_changed == 'true'
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          RELEASE_TAG: ${{ env.RELEASE_TAG }}
+        run: |
+          git config --global user.email "41898282+github-actions[bot]@users.noreply.github.com"
```
Comment (Eusebiotrigo) [inline .github/workflows/publish.yml:43]: Who dis?

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+          git config --global user.name "GitHub Actions"
+          
+          git tag ${{ env.NEW_VERSION }}
```
Comment (Eusebiotrigo) [inline .github/workflows/publish.yml:46]: Where is NEW_VERSION declared/defined?

**Review by Eusebiotrigo — CHANGES_REQUESTED** [review]

File: .github/workflows/publish.yml (hunk 1 above)
Comment (froblesmartin) [reply .github/workflows/publish.yml:43]: https://api.github.com/users/github-actions%5Bbot%5D bot user representing GitHub Actions

File: .github/workflows/publish.yml (hunk 2 above)
Comment (froblesmartin) [reply .github/workflows/publish.yml]: Good catch! :) Fixed

**Review by racuevji — APPROVED** [review]

**Review by Eusebiotrigo — APPROVED** [review]

**Pull request merged**